File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/usecases/PlaceOrderVerification.java
package com.mycompany.myshop.testkit.dsl.core.usecase.usecases;

import com.mycompany.myshop.testkit.driver.port.dtos.PlaceOrderResponse;
import com.mycompany.myshop.testkit.dsl.core.shared.ResponseVerification;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;

import static org.assertj.core.api.Assertions.assertThat;

public class PlaceOrderVerification extends ResponseVerification<PlaceOrderResponse> {
    public PlaceOrderVerification(PlaceOrderResponse response, UseCaseContext context) {
        super(response, context);
    }

    public PlaceOrderVerification orderNumber(String orderNumberResultAlias) {
        var expectedOrderNumber = getContext().getResultValue(orderNumberResultAlias);
        var actualOrderNumber = getResponse().getOrderNumber();
        assertThat(actualOrderNumber)
                .withFailMessage("Expected order number to be '%s', but was '%s'", expectedOrderNumber, actualOrderNumber)
                .isEqualTo(expectedOrderNumber);
        return this;
    }

    public PlaceOrderVerification orderNumberStartsWith(String prefix) {
        var actualOrderNumber = getResponse().getOrderNumber();
        assertThat(actualOrderNumber)
                .withFailMessage("Expected order number to start with '%s', but was '%s'", prefix, actualOrderNumber)
                .startsWith(prefix);
        return this;
    }
}

File: system-test/dotnet/Dsl.Core/UseCase/UseCases/PlaceOrderVerification.cs
using Dsl.Core.Shared;
using Shouldly;
using Driver.Port.Dtos;

namespace Dsl.Core.UseCase.UseCases;

public class PlaceOrderVerification : ResponseVerification<PlaceOrderResponse>
{
    public PlaceOrderVerification(PlaceOrderResponse response, UseCaseContext context)
        : base(response, context)
    {
    }

    public PlaceOrderVerification OrderNumber(string orderNumberResultAlias)
    {
        var expectedOrderNumber = Context.GetResultValue(orderNumberResultAlias);
        var actualOrderNumber = Response.OrderNumber;

        actualOrderNumber.ShouldBe(expectedOrderNumber, $"Expected order number to be '{expectedOrderNumber}', but was '{actualOrderNumber}'");

        return this;
    }

    public PlaceOrderVerification OrderNumberStartsWith(string prefix)
    {
        Response.OrderNumber.ShouldStartWith(prefix, Case.Sensitive, $"Expected order number to start with '{prefix}', but was '{Response.OrderNumber}'");
        return this;
    }
}



